pkg/services/alerting/alerting_usage.go

```diff
@@ -0,0 +1,111 @@
+package alerting
+
+import (
+	"encoding/json"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// DatasourceAlertUsage is a hash where the key represents the
+// Datasource type and the value represents how many alerts
+// that use the datasources.
+type DatasourceAlertUsage map[string]int
+
+// UsageStats contains stats about alert rules configured in
+// Grafana.
+type UsageStats struct {
+	DatasourceUsage DatasourceAlertUsage
+}
+
+// UsageStatsQuerier returns usage stats about alert rules
+// configured in Grafana.
+type UsageStatsQuerier interface {
+	QueryUsageStats() (*UsageStats, error)
+}
+
+// QueryUsageStats returns usage stats about alert rules
+// configured in Grafana.
+func (ae *AlertEngine) QueryUsageStats() (*UsageStats, error) {
+	cmd := &models.GetAllAlertsQuery{}
+	err := ae.Bus.Dispatch(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	dsUsage, err := ae.mapRulesToUsageStats(cmd.Result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &UsageStats{
+		DatasourceUsage: dsUsage,
+	}, nil
+}
+
+func (ae *AlertEngine) mapRulesToUsageStats(rules []*models.Alert) (DatasourceAlertUsage, error) {
+	// map of datasourceId type and frequency
+	typeCount := map[int64]int{}
+
+	for _, a := range rules {
+		dss, err := ae.parseAlertRuleModel(a.Settings)
+		if err != nil {
+			ae.log.Debug("could not parse settings for alert rule", "id", a.Id)
+			continue
+		}
+
+		for _, d := range dss {
+			typeCount[d]++
+		}
+	}
+
+	result := map[string]int{}
+	for k, v := range typeCount {
+		query := &models.GetDataSourceByIdQuery{Id: k}
+		err := ae.Bus.Dispatch(query)
+		if err != nil {
+			return map[string]int{}, nil
+		}
+
+		result[query.Result.Type] = v
+	}
+
+	return result, nil
+}
+
+func (ae *AlertEngine) parseAlertRuleModel(settings *simplejson.Json) ([]int64, error) {
+	datasourceIDs := []int64{}
+	model := alertJSONModel{}
+
+	if settings == nil {
+		return datasourceIDs, nil
+	}
+
+	bytes, err := settings.MarshalJSON()
+	if err != nil {
+		return datasourceIDs, err
+	}
+
+	err = json.Unmarshal(bytes, &model)
+	if err != nil {
+		return datasourceIDs, err
+	}
```

Comment:
Model have already been unmarshaled to simplejson, now we marshal it again and then unmarshal into struct. Potential problem having many rules?

Reply:
Doubt it will be any problems considering how small alert.settings content is. 
Ex https://github.com/grafana/grafana/pull/23125/files#diff-b453ae259d13a0de1e3de4a798638bf9R2

But if getting the byte array in a more performant way would be preferable.

Reply:
An alternative solution would be to not use the `models.Alert` struct to fetch the alert rules but I don't like introducing a alert rule struct that uses `[]byte` as settings just for this use-case that use. The same could be said about the json structs but overall I think we should use structs over simplejson when possible.

Reply:
Yeah maybe I'm overreacting.